natlas-server/tests/cli/test_scope_cli.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+from app.cli.scope import import_items
+from app.models import ScopeItem
+
+
+def test_import_items_no_flags(runner):
+    with runner.isolated_filesystem():
+        with open("scope.txt", "w") as f:
+            scope_items = ["10.0.0.0/8", "192.168.1.0/24", "192.168.5.0/28"]
+            f.write("\n".join(scope_items))
+        result = runner.invoke(import_items, ["scope.txt"])
+        assert result.exit_code == 0
+        imported_scope = [
+            item.target for item in ScopeItem.query.filter_by(blacklist=False).all()
+        ]
+        assert scope_items == imported_scope
+        assert json.loads(result.output)
+
+
+def test_import_items_scope_flag(runner):
+    with runner.isolated_filesystem():
+        with open("scope.txt", "w") as f:
+            scope_items = ["10.0.0.0/8", "192.168.1.0/24", "192.168.5.0/28"]
+            f.write("\n".join(scope_items))
+        result = runner.invoke(import_items, ["--scope", "scope.txt"])
+        assert result.exit_code == 0
+        imported_scope = [
+            item.target for item in ScopeItem.query.filter_by(blacklist=False).all()
+        ]
+        assert scope_items == imported_scope
+        assert json.loads(result.output)
+
+
+def test_import_items_blacklist_flag(runner):
+    with runner.isolated_filesystem():
+        with open("scope.txt", "w") as f:
+            scope_items = ["10.0.0.0/8", "192.168.1.0/24", "192.168.5.0/28"]
+            f.write("\n".join(scope_items))
+        result = runner.invoke(import_items, ["--blacklist", "scope.txt"])
+        assert result.exit_code == 0
+        imported_scope = [
+            item.target for item in ScopeItem.query.filter_by(blacklist=True).all()
+        ]
+        assert scope_items == imported_scope
+        assert json.loads(result.output)
+
+
+def test_import_items_verbose(runner):
+    with runner.isolated_filesystem():
+        with open("scope.txt", "w") as f:
+            scope_items = ["10.0.0.0/8", "192.168.1.0/24", "192.168.5.0/28"]
+            f.write("\n".join(scope_items))
```

Comment:
How about a method that mocks your file for you:
```python
mock_scope_file(*items: string) -> string:
   with open("scope.txt", "w") as f:
            scope_items = items
            f.write("\n".join(scope_items))
   return "scope.txt"

def test_stuff():
   with isolate:
     name = mock_scope_file(DEFAULT_ITEMS)
     result = runner.invoke(import_items, ["--verbose", name])
```

Reply:
Ahh that's a good idea, I knew I was repeating myself too many times but the context manager stuff made me ignore it without realizing that the inner context could easily be extracted.